src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreConfiguredInputDataSource.cs

```diff
@@ -58,6 +64,8 @@ protected override IDisposable LinkExternalInput(ITargetBlock<RestoreUpdate> tar
             // and someone is blocked on us on the same thread, the call proceeds
             JoinUpstreamDataSources(source);
 
+            _packageReferenceTelemetryService.LogPackageRestoreEvent(new PackageRestoreTelemetryEvent(PackageRestoreOperationNames.PackageRestoreConfiguredInputDataSourceLinkedToExternalInput, ContainingProject.FullPath));
```

Comment:
Most calls to these `Log*Event` methods seem to be passing a value that's derived from either the `ConfiguredProject` or `UnconfiguredProject`. Consider whether you can move the `IPackageRestoreTelemetryService`/`ILanguageServiceTelemetryService` interfaces into one of those scopes and have their implementations import the relevant projects directly, so that callers don't need to pass them on every call.